spec/akasha/event_router_spec.rb

```diff
@@ -7,16 +7,26 @@
     Akasha::Aggregate.connect!(repo)
   end
 
-  describe '#connect' do
-    before do
-      subject.register_event_listener(:name_changed, Notifier)
-    end
-
+  shared_examples_for 'registred actions' do
     it "routes repo's events" do
       subject.connect!(repo)
       item.name = 'new name'
       expect_any_instance_of(Notifier).to receive(:on_name_changed).with('item-1', old_name: nil, new_name: 'new name')
       repo.save_aggregate(item)
     end
   end
+
+  describe '#connect' do
+    before do
+      subject.register_event_listener(:name_changed, Notifier)
+    end
+
+    include_examples 'registred actions'
+  end
+
+  context 'when passing single handler for event via constructor' do
```

Comment:
This should go to tests for `EventListenerBase`. All this test does is it verifies that the listener subscribes to repo.

Reply:
Ok, We should extract test suite for router's base and use it on every child class.

Reply:
Makes sense. 👍